### PR TITLE
Opt in notifications

### PR DESCRIPTION
**Pull request by rodericj**
notificationToggle

### CODE REVIEW

File: SF iOS/SF iOS/Feed/EventsFeedViewController.m [hunk 1]
```diff
@@ -66,6 +69,84 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
...
+        };
+    };
+    [alert addAction:[UIAlertAction actionWithTitle:buttonTitle
+                                              style:UIAlertActionStyleDefault
+                                            handler:notificationsPreferenceChangeTapped]];
+
+    [alert addAction:[UIAlertAction actionWithTitle:cancelButtonTitle
+                                              style:UIAlertActionStyleCancel
+                                            handler:nil]];
+    [self presentViewController:alert animated:true completion:nil];
```
Comment (mcritz) [inline SF iOS/SF iOS/Feed/EventsFeedViewController.m:150]: I’d like the UI layering cleaned up a little, please.

Before the alert is show, hide the button. On completion, show the button again.

File: SF iOS/SF iOS/Assets.xcassets/button - subscribed.imageset/Contents.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
+    {
+      "idiom" : "universal",
+      "filename" : "button - subscribed.pdf",
```
Comment (mcritz) [inline SF iOS/SF iOS/Assets.xcassets/button - subscribed.imageset/Contents.json:5]: I’m worried these buttons are really clunky looking and want to tweak them a bit to be more beautiful.

**Review by mcritz — CHANGES_REQUESTED** [review]

File: SF iOS/SF iOS/Feed/EventsFeedViewController.m (hunk 1 above)
Comment (rodericj) [reply SF iOS/SF iOS/Feed/EventsFeedViewController.m:150]: Not sure if I follow. Which button are we talking about? The `UIAction` button or the notification button that was added in this PR? And is the alert you are speaking of the Action Sheet or the system alert asking for notification permissions?

File: SF iOS/SF iOS/Assets.xcassets/button - subscribed.imageset/Contents.json (hunk 2 above)
Comment (rodericj) [reply SF iOS/SF iOS/Assets.xcassets/button - subscribed.imageset/Contents.json:5]: I'm open to it.

File: SF iOS/SF iOS/Feed/EventsFeedViewController.m (hunk 1 above)
Comment (mcritz) [reply SF iOS/SF iOS/Feed/EventsFeedViewController.m:150]: Notification button should hide before the action sheet appears and return when the action sheet dismisses.

Comment (rodericj) [reply SF iOS/SF iOS/Feed/EventsFeedViewController.m:150]: Got it

Comment (rodericj) [reply SF iOS/SF iOS/Feed/EventsFeedViewController.m:150]: Ok yeah that is a lot better. I pushed the change just now.

**Comment by rodericj** [comment]
This is merged into #74

**Pull request closed without merging**